runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -249,3 +249,68 @@ J9::ARM64::TreeEvaluator::arraylengthEvaluator(TR::Node *node, TR::CodeGenerator
 
    return lengthReg;
    }
+
+TR::Register *
+J9::ARM64::TreeEvaluator::BNDCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *src1Reg;
+   TR::Register *src2Reg = NULL;
+   int32_t value;
+   TR::LabelSymbol *snippetLabel;
+   TR::Instruction *gcPoint;
+   bool reversed = false;
+
+   if ((firstChild->getOpCode().isLoadConst())
+         && (0x80000000 >= firstChild->getInt())
+         && (NULL == firstChild->getRegister()))
+      {
+      src2Reg = cg->evaluate(secondChild);
+      reversed = true;
+      }
+   else
+      {
+      src1Reg = cg->evaluate(firstChild);
+
+      if ((secondChild->getOpCode().isLoadConst())
+            && (NULL == secondChild->getRegister()))
+         {
+         value = secondChild->getInt();
+         if (0 > value || 0x80000000 < value)
```

Comment:
I think you want to evaluate the child into a register only if the value is outside the 12-bit immediate range of the instruction.  This test is backwards.  See `constantIsUnsignedImm12`.